### ISSUE TITLE
Cop Naming/FileName not honoring "Exclude" property

### ISSUE THREAD

**Issue by molson504x** [rubocop/rubocop#13770]
When I'm telling the Naming/FileName cop to Exclude files (either through AllCops config or through the Naming/FileName cop config) it does not exclude files.  [Per the docs](https://docs.rubocop.org/rubocop/1.68/cops_naming.html#configurable-attributes-namingfilename) this should work.

--------

## Expected behavior

When I have a rubocop config file containing either:

``` yaml
AllCops:
    TargetRubyVersion: 3
    Exclude:
        - 'bb_bin/someRubyScript.rb' 
```

or...

``` yaml
Naming/FileName:
    Exclude:
        - 'bb_bin/someRubyScript.rb' 
```

the file should be ignored.

## Actual behavior

The file is not ignored in rubocop results:

``` text
  Inspecting 6 files
  ...C..
  
  Offenses:
  
  bb_bin/someRubyScript.rb:1:1: C: Naming/FileName: The name of this source file (someRubyScript.rb) should use snake_case.
  
  6 files inspected, 1 offense detected
```

## Steps to reproduce the problem

1. Have a repo containing a file that doesn't follow snake_case
2. Create a rule similar to above...
3. Run Rubocop.  Observe results.

## RuboCop version

1.68.0

I'm using SuperLinter, not RuboCop directly.  Here's the gems used in SuperLinter:

``` ruby
gem "rubocop", "~> 1.68.0"
gem "rubocop-github", "~> 0.20.0"
gem "rubocop-minitest", "~> 0.36.0"
gem "rubocop-performance", "~>1.22.1"
gem "rubocop-rails", "~> 2.27"
gem "rubocop-rake", "~> 0.6.0"
gem "rubocop-rspec", "~> 3.2.0"
gem "standard", "~>1.42"
gem "rubocop-capybara", "~> 2.21"
gem "rubocop-factory_bot", "~> 2.26"
gem "rubocop-rspec_rails", "~> 2.30"
```

**Comment by molson504x** [comment]
Here's what I did to reproduce locally:

1. Install all of the gems listed above (lots of `gem install` commands)
2. In my ruby project, add a ruby file named "someRubyScript.rb"
3. In my rubocop config, add the exclusion criteria for the `Naming/FileName` cop
4. Run `rubocop -c '.github/linters/.rubocop.yml' --force-exclusion --ignore-unrecognized-cops`

Observe the following message in the output:
> bb_bin/someRubyScript.rb:1:1: C: Naming/FileName: The name of this source file (someRubyScript.rb) should use snake_case.

Here's my full .rubocop.yml file:

``` yaml
inherit_gem:
  rubocop-github:
    - config/default.yml

AllCops:
    TargetRubyVersion: 3
    Exclude:
        - 'bb_bin/someRubyScript.rb'
Metrics/MethodLength:
    Enabled: false
Layout/LineLength:
    Enabled: false
Style/CommentedKeyword:
    Enabled: false

# Eventually we want to enable this cop and fix all the offenses
Style/GlobalVars:
    Enabled: false
# Eventually we want to enable this cop and fix all the offenses
Metrics/AbcSize:
    Enabled: false
# Eventually we want to enable this cop and fix all the offenses
Naming/AccessorMethodName:
    Enabled: false
Naming/FileName:
    Exclude:
        - 'bb_bin/someRubyScript.rb'
```

Thanks :)

**Comment by molson504x** [comment]
Interesting, so even though I'm running rubocop and passing it the relative path to the config, the config has to then be set to paths relative to itself?

**Comment by bbatsov** [comment]
For the record - originally we had the strong assumption that the (main) configuration for for any particular project would be in the root directory of the project. (with configuration files always nested in general - e.g. you have some in your `$HOME`, in your project, in a subdirectory of the project, etc)
